### PR TITLE
Fix some slice append usages

### PR DESCRIPTION
**Pull request by harryzcy**

### CODE REVIEW

File: routers/api/v1/repo/pull.go [hunk 1]
```diff
@@ -326,11 +326,9 @@ func CreatePullRequest(ctx *context.APIContext) {
 			return
 		}
 
-		labelIDs = make([]int64, len(form.Labels))
-		orgLabelIDs := make([]int64, len(form.Labels))
-
-		for i := range labels {
-			labelIDs[i] = labels[i].ID
+		labelIDs = make([]int64, 0, len(form.Labels))
```
Comment (delvh) [inline routers/api/v1/repo/pull.go]:
```suggestion
		labelIDs = make([]int64, 0, len(labels))
```

File: routers/api/v1/repo/pull.go [hunk 2]
```diff
@@ -340,12 +338,12 @@ func CreatePullRequest(ctx *context.APIContext) {
 				return
 			}
 
-			for i := range orgLabels {
-				orgLabelIDs[i] = orgLabels[i].ID
+			orgLabelIDs := make([]int64, len(form.Labels))
+			for i, orgLabel := range orgLabels {
+				orgLabelIDs[i] = orgLabel.ID
```
Comment (delvh) [inline routers/api/v1/repo/pull.go]:
```suggestion
			orgLabelIDs := make([]int64, 0, len(orgLabels))
			for _, orgLabel := range orgLabels {
				orgLabelIDs = append(orgLabelIDs, orgLabel.ID)
```

**Review by wxiaoguang — APPROVED** [review]
Off-topic: in most cases, there is no need to play with `make(slice, 0, len)` since Gitea is not a time-critical high performance server and most code is not on the hot path, using it or not, it at most only affects nanosecond-level (10^-9) performance, while a normal page request is around 10-200 millisecond (10^-3).

**Review by CaiCandong — APPROVED** [review]

**Comment by KN4CK3R** [comment]
> Off-topic: in most cases, there is no need to play with `make(slice, 0, len)` since Gitea is not a time-critical high performance server and most code is not on the hot path, using it or not, it at most only affects nanosecond-level (10^-9) performance, while a normal page request is around 10-200 millisecond (10^-3).

But here it's not a performance improvement but a bugfix because the slices a prepended with default elements.
```go
s := make([]string, 5)
s = append(s, "foo") // []string{"", "", "", "", "", "foo"}
```

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> > Off-topic: in most cases, there is no need to play with `make(slice, 0, len)` since Gitea is not a time-critical high performance server and most code is not on the hot path, using it or not, it at most only affects nanosecond-level (10^-9) performance, while a normal page request is around 10-200 millisecond (10^-3).
> 
> But here it's not a performance improvement but a bugfix because the slices a prepended with default elements.
> 
> ```go
> s := make([]string, 5)
> s = append(s, "foo") // []string{"", "", "", "", "", "foo"}
> ```

Yup, I know, so I said "off-topic"

Actually, force using "make(...)" would cause more similar bugs. `var slice []int` is always safe and can't be wrong.

**Pull request merged**